GenericJSON/Querying.swift

```diff
@@ -74,4 +74,33 @@ public extension JSON {
     subscript(dynamicMember member: String) -> JSON? {
         return self[member]
     }
+    
+    /// Return the JSON type at the keypath if this is an `.object`, otherwise `nil`
+    ///
+    /// This lets you write `json["foo.bar.jar"]`.
```

Comment:
This isn’t really true, is it? Because writing `json["foo.bar.jar"]` would mean going through the `subscript(key:)` function, and that one doesn’t support the key path syntax. But I wonder, could we simply move the “key path” support to `subscript(key:)`, so that `json["foo.bar.jar"]` would work as expected?

Reply:
Yes, you are right. It was a copy-paste error. It should read:
"This lets you write `json.valueForKeyPath("foo.bar.jar")`"

Good point, I thought about adding "key path" functionality directly to `subscript(key:)` but was concerned about accidental key path look ups for keys which just *happened* to contain a period (as you pointed out).

My first idea was to allow the consumer to specify they wanted key path look up explicitly through an `iskeyPath` bool which defaults to `false`. However subscripts are not permitted to have default values. 

```swift
public subscript(key: String, isKeyPath: Bool = false) -> JSON? { // Error: Default arguments are not allowed in subscripts
   
}
```

Dropping the default makes the api awkward especially if it resides next to `subscript(key:)`. Supplying `false` to the function below is the same as `subscript(key:)` and the duplication doesn't sit well with me. What are your thoughts?

```swift
public subscript(key: String, isKeyPath: Bool) -> JSON? { 
        // ...
}
```